arch/x86/core/ia32/intstub.S

```diff
@@ -174,6 +174,12 @@ alreadyOnIntStack:
 	cli			/* disable interrupts again */
 #endif
 
+#if defined (CONFIG_SMHI_BSP)
```

Comment:
where is this config defined and what does it mean? The kconfig needs to be defined in Zephyr and enabled/used by some platform. We should not depend on Kconfigs defined out of tree.

Reply:
it's defined in ISH SoC & Board layer which is not upstream,  since this config is platform(ISH) specific,  what's your idea for this ?  thanks.

Reply:
is it really ish specific? Could this feature be tied to x86 or lakemont instead and made generic and not soc/platform specific? btw, where is `set_irq_stats` defined?

Reply:
Just refine that patch to make it more generic, please help review again, thank you very much :)